### DISCUSSION TITLE
Signup with email confirmations returns null (Flutter)

### DISCUSSION THREAD

**Discussion by bhavinks** [supabase/supabase#5357] [Questions] [answered]
Hello, I am testing an app where I have a registration screen for email signup. According to the documentation, once user is submitted via the signup API, it returns a user object and no session object until the user has confirmed their email. However for me it is returning a null regardless of whether its a valid user or an invalid user (email is already registered). If I turn off the email verification, then it returns a user object and session object immediately. Is this a known bug or am I doing something wrong ? Code sample below.

```dart
RoundedButton(
  buttonColor: Colors.blueAccent,
  buttonTitle: 'Register',
  onPressed: () async {
    setState(() {
      showSpinner = true;
    });
    String err = "";
    // try {
    final response = await Supabase.instance.client.auth
        .signUp(_email, _password);
    print(response.user?.id);
    setState(() {
      showSpinner = false;
});
```

**Comment by dshukertjr** [comment, marked as answer]
Hi @bhavinks,

Sorry for the late response here. Yes, this was a known bug, and it has been fixed in the recent versions of Supabase-dart and Supabase-flutter!
